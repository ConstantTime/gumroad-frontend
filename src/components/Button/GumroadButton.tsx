import Button from "@material-ui/core/Button";
import React from "react";
import { IGumroadButton } from "./types";

const GumroadButton: React.FC<IGumroadButton> = ({ onClick, children }) => {
  return (
    <Button
      style={{ backgroundColor: "#FFFFFF", color: "#000000" }}
      size="medium"
      variant="contained"
      onClick={onClick}
    >
      {children}
    </Button>
  );
};
export default GumroadButton;
