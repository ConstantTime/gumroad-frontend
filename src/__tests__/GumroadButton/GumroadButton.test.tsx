import { shallow } from "enzyme";
import React from "react";
import GumroadButton from "../../components/Button/GumroadButton";

describe("Gumroad button", () => {
  it("should render the component", () => {
    const wrapper = shallow(<GumroadButton onClick={jest.fn()} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger the onClick handler on clicking button", () => {
    const handleClick = jest.fn();
    const wrapper = shallow(<GumroadButton onClick={handleClick} />);
    wrapper.simulate("click");

    expect(handleClick).toHaveBeenCalled();
  });
});
