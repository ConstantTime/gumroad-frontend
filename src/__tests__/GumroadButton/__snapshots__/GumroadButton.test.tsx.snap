// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gumroad button should render the component 1`] = `
<WithStyles(ForwardRef(Button))
  onClick={[MockFunction]}
  size="medium"
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "color": "#000000",
    }
  }
  variant="contained"
/>
`;
